LazyCache/CachingService.cs

```diff
@@ -141,6 +141,13 @@ public virtual void Remove(string key)
             ValidateKey(key);
             CacheProvider.Remove(key);
         }
+        
+        public virtual void RemoveAll()
+        {
+            CacheProvider.Dispose();
+            cacheProvider = new Lazy<ICacheProvider>(() => 

```

Comment:
Initializing a new `cacheProvider` instead of removing the keys from the existing `cacheProvider` will gives inconsistent behavior to how `Remove(key)` works.
E.g. if the `CachingService` was instantiated using any of the constructors except `CachingService()`, then calling `RemoveAll` will change the underlying `cacheProvider` to a `DefaultCacheProvider`.

Reply:
You are right.
I made this PR because I use dependency injection to create a CachingService, and the PR #91 doesn't work correctly with DI.
I think it's best to abandon this PR, and edit PR #91 so dependency injection code (asp.net core, ninject) works correctly.

Reply:
Yeah this sort of change will need to incorporate DI so will close in favour of #91